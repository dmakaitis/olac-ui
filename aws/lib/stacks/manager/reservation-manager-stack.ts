import * as cdk from 'aws-cdk-lib';
import * as lambda from 'aws-cdk-lib/aws-lambda';
import {Construct} from 'constructs';
import {Function} from 'aws-cdk-lib/aws-lambda';
import {Duration} from "aws-cdk-lib";

export interface ReservationManagerStackProps extends cdk.StackProps {

    saveReservationFunction: Function;

    payPal: {
        apiBase: string,
        clientId: string,
        secret: string
    };

}

export class ReservationManagerStack extends cdk.Stack {

    readonly apiGetNewReservationIdFunction: Function;
    readonly apiSaveReservationFunction: Function;
    readonly apiSaveReservationAdminFunction: Function;

    constructor(scope: Construct, id: string, props: ReservationManagerStackProps) {
        super(scope, id, props);

        this.apiGetNewReservationIdFunction = new lambda.Function(this, 'GetNewReservationId', {
            description: 'Generate a new reservation ID',
            runtime: lambda.Runtime.NODEJS_22_X,
            code: lambda.Code.fromAsset('./dist/lambda/manager/reservation/new-reservation-id'),
            handler: 'new-reservation-id.handler',
        });

        this.apiSaveReservationFunction = new Function(this, 'SaveReservationApi', {
            description: 'Saves a new reservation to the datastore (for public use)',
            runtime: lambda.Runtime.NODEJS_22_X,
            code: lambda.Code.fromAsset('./dist/lambda/manager/reservation/save-reservation'),
            handler: 'save-reservation.apiHandler',
            environment: {
                SAVE_RESERVATION_FUNCTION: props.saveReservationFunction.functionName,
                PAYPAL_API_BASE: props.payPal.apiBase,
                PAYPAL_CLIENT_ID: props.payPal.clientId,
                PAYPAL_SECRET: props.payPal.secret
            },
            timeout: Duration.seconds(10)
        });
        props.saveReservationFunction.grantInvoke(this.apiSaveReservationFunction);

        this.apiSaveReservationAdminFunction = new Function(this, 'SaveReservationAdminApi', {
            description: 'Saves a new reservation to the datastore (for event coordinator use)',
            runtime: lambda.Runtime.NODEJS_22_X,
            code: lambda.Code.fromAsset('./dist/lambda/manager/reservation/save-reservation-admin'),
            handler: 'save-reservation-admin.apiHandler',
            environment: {
                SAVE_RESERVATION_FUNCTION: props.saveReservationFunction.functionName,
                PAYPAL_API_BASE: props.payPal.apiBase,
                PAYPAL_CLIENT_ID: props.payPal.clientId,
                PAYPAL_SECRET: props.payPal.secret
            }
        });
        props.saveReservationFunction.grantInvoke(this.apiSaveReservationAdminFunction);
    }
}