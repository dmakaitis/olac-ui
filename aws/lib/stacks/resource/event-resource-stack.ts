import * as cdk from 'aws-cdk-lib';
import * as ddb from 'aws-cdk-lib/aws-dynamodb';
import {BillingMode} from 'aws-cdk-lib/aws-dynamodb';
import * as lambda from 'aws-cdk-lib/aws-lambda';
import {Construct} from 'constructs';

export class EventResourceStack extends cdk.Stack {

    listEventsFunction: cdk.aws_lambda.Function;
    saveEventFunction: cdk.aws_lambda.Function;
    deleteEventFunction: cdk.aws_lambda.Function;

    listReservationsFunction: cdk.aws_lambda.Function;
    listReservationsCsvFunction: cdk.aws_lambda.Function;
    saveReservationFunction: cdk.aws_lambda.Function;
    deleteReservationFunction: cdk.aws_lambda.Function;

    areTicketsAvailableFunction: cdk.aws_lambda.Function;

    eventsTable: cdk.aws_dynamodb.Table;
    auditTable: cdk.aws_dynamodb.Table;

    constructor(scope: Construct, id: string, props: cdk.StackProps) {
        super(scope, id, props);
        this.defineEventFunctions();
        this.defineReservationFunctions();
    }

    private defineEventFunctions() {
        const table = new ddb.Table(this, 'EventTable', {
            partitionKey: {name: 'id', type: ddb.AttributeType.STRING},
            billingMode: BillingMode.PAY_PER_REQUEST
        });
        table.addGlobalSecondaryIndex({
            indexName: 'GlobalSortByEventDateIndex',
            partitionKey: {name: 'index', type: ddb.AttributeType.STRING},
            sortKey: {name: 'eventDate', type: ddb.AttributeType.STRING}
        });
        this.eventsTable = table;

        this.listEventsFunction = new lambda.Function(this, 'ListEvents', {
            description: 'Retrieves the list of OLAC events from the datastore',
            runtime: lambda.Runtime.NODEJS_22_X,
            code: lambda.Code.fromAsset('./lambda/resource/event'),
            handler: 'get-events.handler',
            environment: {
                TABLE_NAME: table.tableName
            },
        });
        table.grantReadData(this.listEventsFunction);

        this.saveEventFunction = new lambda.Function(this, 'SaveEvent', {
            description: 'Saves OLAC event information to the datastore',
            runtime: lambda.Runtime.NODEJS_22_X,
            code: lambda.Code.fromAsset('./lambda/resource/event'),
            handler: 'save-event.handler',
            environment: {
                TABLE_NAME: table.tableName
            },
        });
        table.grantReadWriteData(this.saveEventFunction);

        this.deleteEventFunction = new lambda.Function(this, 'DeleteEvent', {
            description: 'Deletes OLAC event information from the datastore',
            runtime: lambda.Runtime.NODEJS_22_X,
            code: lambda.Code.fromAsset('./lambda/resource/event'),
            handler: 'delete-event.handler',
            environment: {
                TABLE_NAME: table.tableName
            },
        });
        table.grantReadWriteData(this.deleteEventFunction);
    }

    private defineReservationFunctions() {
        const table = new ddb.Table(this, 'ReservationTable', {
            partitionKey: {name: 'id', type: ddb.AttributeType.STRING},
            billingMode: BillingMode.PAY_PER_REQUEST
        });
        table.addGlobalSecondaryIndex({
            indexName: 'GlobalEventIndex',
            partitionKey: {name: 'eventId', type: ddb.AttributeType.STRING},
            sortKey: {name: 'reservationId', type: ddb.AttributeType.NUMBER}
        });

        this.auditTable = new ddb.Table(this, 'ReservationAuditTable', {
            partitionKey: {name: 'id', type: ddb.AttributeType.STRING},
            billingMode: BillingMode.PAY_PER_REQUEST
        });
        this.auditTable.addGlobalSecondaryIndex({
            indexName: 'GlobalReservationIndex',
            partitionKey: {name: 'reservationId', type: ddb.AttributeType.STRING},
            sortKey: {name: 'timestamp', type: ddb.AttributeType.STRING}
        });

        const counterTable = new ddb.Table(this, 'ReservationCounterTable', {
            partitionKey: {name: 'id', type: ddb.AttributeType.STRING},
            billingMode: BillingMode.PAY_PER_REQUEST
        });

        this.listReservationsFunction = new lambda.Function(this, 'ListEventReservations', {
            description: 'Retrieves the list of OLAC event reservations from the datastore',
            runtime: lambda.Runtime.NODEJS_22_X,
            code: lambda.Code.fromAsset('./lambda/resource/event'),
            handler: 'get-event-reservations.handler',
            environment: {
                TABLE_NAME: table.tableName
            },
        });
        table.grantReadData(this.listReservationsFunction);

        this.listReservationsCsvFunction = new lambda.Function(this, 'ListEventReservationsCsv', {
            description: 'Retrieves the list of OLAC event reservations from the datastore in CSV format',
            runtime: lambda.Runtime.NODEJS_22_X,
            code: lambda.Code.fromAsset('./lambda/resource/event'),
            handler: 'get-event-reservations-csv.handler',
            environment: {
                TABLE_NAME: table.tableName
            },
        });
        table.grantReadData(this.listReservationsCsvFunction);

        this.saveReservationFunction = new lambda.Function(this, 'SaveEventReservation', {
            description: 'Saves OLAC event reservation information to the datastore',
            runtime: lambda.Runtime.NODEJS_22_X,
            code: lambda.Code.fromAsset('./lambda/resource/event'),
            handler: 'save-reservation.handler',
            environment: {
                TABLE_NAME: table.tableName,
                COUNTER_TABLE_NAME: counterTable.tableName,
                AUDIT_TABLE_NAME: this.auditTable.tableName
            },
        });
        table.grantReadWriteData(this.saveReservationFunction);
        counterTable.grantReadWriteData(this.saveReservationFunction);
        this.auditTable.grantWriteData(this.saveReservationFunction);

        this.deleteReservationFunction = new lambda.Function(this, 'DeleteReservation', {
            description: 'Deletes an OLAC event reservation from the datastore',
            runtime: lambda.Runtime.NODEJS_22_X,
            code: lambda.Code.fromAsset('./lambda/resource/event'),
            handler: 'delete-reservation.handler',
            environment: {
                TABLE_NAME: table.tableName
            },
        });
        table.grantReadWriteData(this.deleteReservationFunction);

        this.areTicketsAvailableFunction = new lambda.Function(this, 'AreTicketsAvailable', {
            description: 'Checks to see if the requested number of tickets are available for an event',
            runtime: lambda.Runtime.NODEJS_22_X,
            code: lambda.Code.fromAsset('./lambda/resource/event'),
            handler: 'are-tickets-available.apiHandler',
            environment: {
                EVENT_TABLE_NAME: this.eventsTable.tableName,
                RESERVATION_TABLE_NAME: table.tableName
            }
        });
        this.eventsTable.grantReadData(this.areTicketsAvailableFunction);
        table.grantReadData(this.areTicketsAvailableFunction);
    }
}