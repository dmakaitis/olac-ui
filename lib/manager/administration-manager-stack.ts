import * as cdk from 'aws-cdk-lib';
import * as lambda from 'aws-cdk-lib/aws-lambda';
import {Construct} from 'constructs';

export interface AdministrationManagerStackProps extends cdk.StackProps {
    getConfigFunction: lambda.Function
}

export class AdministrationManagerStack extends cdk.Stack {

    readonly getClientConfigFunction: lambda.Function;

    constructor(scope: Construct, id: string, props: AdministrationManagerStackProps) {
        super(scope, id, props);

        this.getClientConfigFunction = new lambda.Function(this, 'GetClientConfig', {
            description: 'Retrieve the client configuration for the UI',
            runtime: lambda.Runtime.NODEJS_16_X,
            code: lambda.Code.fromAsset('./lambda/manager/administration'),
            handler: 'get-client-config.handler',
            environment: {
                GET_CONFIG_FUNCTION: props.getConfigFunction.functionName
            },
        });

        props.getConfigFunction.grantInvoke(this.getClientConfigFunction);
    }
}