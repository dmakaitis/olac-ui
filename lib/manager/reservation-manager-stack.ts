import * as cdk from 'aws-cdk-lib';
import * as lambda from 'aws-cdk-lib/aws-lambda';
import {Construct} from 'constructs';
import {Function} from 'aws-cdk-lib/aws-lambda';

export interface ReservationManagerStackProps extends cdk.StackProps {

    saveReservationFunction: Function

}

export class ReservationManagerStack extends cdk.Stack {

    readonly apiGetNewReservationIdFunction: Function;
    readonly apiSaveReservationFunction: Function;
    readonly apiSaveReservationAdminFunction: Function;

    constructor(scope: Construct, id: string, props: ReservationManagerStackProps) {
        super(scope, id, props);

        this.apiGetNewReservationIdFunction = new lambda.Function(this, 'GetNewReservationId', {
            description: 'Generate a new reservation ID',
            runtime: lambda.Runtime.NODEJS_16_X,
            code: lambda.Code.fromAsset('./lambda/manager/reservation'),
            handler: 'new-reservation-id.handler',
        });

        this.apiSaveReservationFunction = new Function(this, 'SaveReservationApi', {
            description: 'Saves a new reservation to the datastore (for public use)',
            runtime: lambda.Runtime.NODEJS_16_X,
            code: lambda.Code.fromAsset('./lambda/manager/reservation'),
            handler: 'save-reservation.apiHandler',
            environment: {
                SAVE_RESERVATION_FUNCTION: props.saveReservationFunction.functionName
            }
        });
        props.saveReservationFunction.grantInvoke(this.apiSaveReservationFunction);

        this.apiSaveReservationAdminFunction = new Function(this, 'SaveReservationAdminApi', {
            description: 'Saves a new reservation to the datastore (for event coordinator use)',
            runtime: lambda.Runtime.NODEJS_16_X,
            code: lambda.Code.fromAsset('./lambda/manager/reservation'),
            handler: 'save-reservation-admin.apiHandler',
            environment: {
                SAVE_RESERVATION_FUNCTION: props.saveReservationFunction.functionName
            }
        });
        props.saveReservationFunction.grantInvoke(this.apiSaveReservationAdminFunction);
    }
}