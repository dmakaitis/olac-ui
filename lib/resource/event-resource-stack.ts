import * as cdk from 'aws-cdk-lib';
import * as ddb from 'aws-cdk-lib/aws-dynamodb';
import * as lambda from 'aws-cdk-lib/aws-lambda';
import {Construct} from 'constructs';

export class EventResourceStack extends cdk.Stack {
    constructor(scope: Construct, id: string, props: cdk.StackProps) {
        super(scope, id, props);

        const table = new ddb.Table(this, 'EventTable', {
            partitionKey: {name: 'id', type: ddb.AttributeType.STRING}
        });

        const saveLambda = new lambda.Function(this, 'SaveEvent', {
            description: 'Saves OLAC event information to the datastore',
            runtime: lambda.Runtime.NODEJS_16_X,
            code: lambda.Code.fromAsset('./lambda/resource/event'),
            handler: 'save.handler',
            environment: {
                TABLE_NAME: table.tableName
            },
        });

        table.grantReadWriteData(saveLambda);
    }
}