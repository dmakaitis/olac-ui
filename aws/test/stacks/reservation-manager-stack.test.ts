import {Template} from "aws-cdk-lib/assertions";
import {App} from "aws-cdk-lib";
import {Function} from "aws-cdk-lib/aws-lambda"
import {ReservationManagerStack} from "../../lib/stacks/manager/reservation-manager-stack";
import {EventResourceStack} from "../../lib/stacks/resource/event-resource-stack";

let app: App;
let saveReservationFunction: Function;

describe("ReservationManagerStack", () => {
    beforeEach(() => {
        app = new App();

        const eventStack = new EventResourceStack(app, "EventResourceStack", {});
        saveReservationFunction = eventStack.saveReservationFunction
    })

    test("Config utility stack creation", () => {
        const stack = new ReservationManagerStack(app, "ReservationManagerStack", {
            payPal: {
                apiBase: "https://test.paypal.com/api",
                clientId: "test-paypal-client-id",
                secret: "my-paypal-secret"
            },
            saveReservationFunction: saveReservationFunction
        });
        const template = Template.fromStack(stack);
        expect(template.toJSON()).toMatchSnapshot()///
    })
})
