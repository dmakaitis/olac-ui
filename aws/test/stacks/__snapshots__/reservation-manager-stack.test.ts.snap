// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationManagerStack Reservation manager stack creation 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "GetNewReservationIdDDB46ED0": {
      "DependsOn": [
        "GetNewReservationIdServiceRole6EE55A20",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "a3c316ebffefb3bd3aba96d56ff90f6f89b4bde1b5111b9f8775dca5124f0bb4.zip",
        },
        "Description": "Generate a new reservation ID",
        "Handler": "new-reservation-id.handler",
        "Role": {
          "Fn::GetAtt": [
            "GetNewReservationIdServiceRole6EE55A20",
            "Arn",
          ],
        },
        "Runtime": "nodejs22.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "GetNewReservationIdServiceRole6EE55A20": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "SaveReservationAdminApi4455CA97": {
      "DependsOn": [
        "SaveReservationAdminApiServiceRoleDefaultPolicy009E90D0",
        "SaveReservationAdminApiServiceRoleC90B1719",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "24f3df2b5b3039ce86399d2437e875bb7ffbee695849711378e172253e68b3d6.zip",
        },
        "Description": "Saves a new reservation to the datastore (for event coordinator use)",
        "Environment": {
          "Variables": {
            "PAYPAL_API_BASE": "https://test.paypal.com/api",
            "PAYPAL_CLIENT_ID": "test-paypal-client-id",
            "PAYPAL_SECRET": "my-paypal-secret",
            "SAVE_RESERVATION_FUNCTION": {
              "Fn::ImportValue": "EventResourceStack:ExportsOutputRefSaveEventReservation38592C5B26F9E271",
            },
          },
        },
        "Handler": "save-reservation-admin.apiHandler",
        "Role": {
          "Fn::GetAtt": [
            "SaveReservationAdminApiServiceRoleC90B1719",
            "Arn",
          ],
        },
        "Runtime": "nodejs22.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "SaveReservationAdminApiServiceRoleC90B1719": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "SaveReservationAdminApiServiceRoleDefaultPolicy009E90D0": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::ImportValue": "EventResourceStack:ExportsOutputFnGetAttSaveEventReservation38592C5BArn797C18A2",
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::ImportValue": "EventResourceStack:ExportsOutputFnGetAttSaveEventReservation38592C5BArn797C18A2",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "SaveReservationAdminApiServiceRoleDefaultPolicy009E90D0",
        "Roles": [
          {
            "Ref": "SaveReservationAdminApiServiceRoleC90B1719",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "SaveReservationApi83E02627": {
      "DependsOn": [
        "SaveReservationApiServiceRoleDefaultPolicyF64BDC17",
        "SaveReservationApiServiceRole9C0B18F4",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "12a74f24feda0c7fd97e57aa577d5c8aaa87bade494b50dc3c354a1e7734ceb6.zip",
        },
        "Description": "Saves a new reservation to the datastore (for public use)",
        "Environment": {
          "Variables": {
            "PAYPAL_API_BASE": "https://test.paypal.com/api",
            "PAYPAL_CLIENT_ID": "test-paypal-client-id",
            "PAYPAL_SECRET": "my-paypal-secret",
            "SAVE_RESERVATION_FUNCTION": {
              "Fn::ImportValue": "EventResourceStack:ExportsOutputRefSaveEventReservation38592C5B26F9E271",
            },
          },
        },
        "Handler": "save-reservation.apiHandler",
        "Role": {
          "Fn::GetAtt": [
            "SaveReservationApiServiceRole9C0B18F4",
            "Arn",
          ],
        },
        "Runtime": "nodejs22.x",
        "Timeout": 10,
      },
      "Type": "AWS::Lambda::Function",
    },
    "SaveReservationApiServiceRole9C0B18F4": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "SaveReservationApiServiceRoleDefaultPolicyF64BDC17": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::ImportValue": "EventResourceStack:ExportsOutputFnGetAttSaveEventReservation38592C5BArn797C18A2",
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::ImportValue": "EventResourceStack:ExportsOutputFnGetAttSaveEventReservation38592C5BArn797C18A2",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "SaveReservationApiServiceRoleDefaultPolicyF64BDC17",
        "Roles": [
          {
            "Ref": "SaveReservationApiServiceRole9C0B18F4",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
